opentelemetry-sdk/src/trace/provider.rs

```diff
@@ -68,32 +68,39 @@ use crate::trace::{
 };
 use crate::Resource;
 use crate::{export::trace::SpanExporter, trace::SpanProcessor};
-use once_cell::sync::{Lazy, OnceCell};
 use opentelemetry::trace::TraceError;
 use opentelemetry::InstrumentationScope;
 use opentelemetry::{otel_debug, trace::TraceResult};
 use std::borrow::Cow;
 use std::sync::atomic::{AtomicBool, Ordering};
-use std::sync::Arc;
+use std::sync::{Arc, OnceLock};
 
 use super::IdGenerator;
 
-static PROVIDER_RESOURCE: OnceCell<Resource> = OnceCell::new();
+// TODO Replace with LazyLock once it is stable
```

Comment:
nit: We might not be able to use `LazyLock` for `PROVIDER_RESOURCE` as `LazyLock` doesn’t allow additional inputs to its function after you call [LazyLock::new(|| ...)](https://doc.rust-lang.org/stable/std/sync/struct.LazyLock.html#method.new).

Reply:
good catch :)